### PR TITLE
Poincare half space (continued)

### PR DESCRIPTION
**Pull request by AliceLeBrigant**
Continue with development of Poincare half space.

### CODE REVIEW

File: tests/test_poincare_half_space.py [hunk 1]
```diff
@@ -25,6 +25,14 @@ def test_belongs(self):
         expected = gs.array([True, False])
         self.assertAllClose(result, expected)
 
+    def test_inner_product_vectorization(self):
```
Comment (nguigs) [inline tests/test_poincare_half_space.py:28]: Did you try other vectorization scenarios? E.g. one pair of tangent_vec + one base_point, one tangant_vec_a, multiple tangent_vec_b, one base_point, or multiple pairs of tangent_vecs at one base_point?

**Review by nguigs — COMMENTED** [review]
Great!
@ninamiolane what do you think about the Deep Source error? I thought we had turned off these types of errors?

File: tests/test_poincare_half_space.py (hunk 1 above)
Comment (AliceLeBrigant) [reply tests/test_poincare_half_space.py:28]: Yes they also work. Should I include them in ```test_poincare_half_space```?

**Review by ninamiolane — APPROVED** [review]
🎉 Awesome, LGTM! Yes, let's ignore the deep source error, this is one that we have chosen to ignore. I tried to change the deep source settings, but couldn't for some reason. They might have changed their interface. I'll try later also, but in the meantime we can merge this!

**Pull request merged**